### PR TITLE
Login logout flow

### PR DESCRIPTION
**Pull request by nurmerey**
- [x] Changing the cookie library to js-cookie
- [x] Refactored API calls
- [x] Login/logout flow done
- [x] Reimplemented Project API
- [x] Contact and Github URL buttons working

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -1,39 +1,29 @@
...
-      "http://127.0.0.1:8000/projects/logout/",
-    );
-    console.log("logout result", result);
-    this.context.fetchData();
+    this.context.logout();
   }
 
   render() {
     const user = this.context.user;
+    console.log("XXXXXX removed auth_token from cookies and reset user", user);
```
Comment (partiallymisplaced) [inline src/components/Header.tsx]: 🗑

File: src/components/ProjectCard.tsx [hunk 2]
```diff
@@ -60,17 +60,17 @@ const ProjectCard: FC<Props> = ({ data }) => {
...
 
       <div className="card-buttons">
-        <a 
-          href={data.githubUrl} 
-          className="button-link" 
-          target="_blank"
+        <a
+          href={data.github_url}
+          className="button-link"
+          target={data.github_url}
```
Comment (partiallymisplaced) [inline src/components/ProjectCard.tsx]: Will this open in the same tab? Same tab vs. new tab?

File: src/components/ProjectCard.tsx [hunk 3]
```diff
@@ -60,17 +60,17 @@ const ProjectCard: FC<Props> = ({ data }) => {
...
           rel="noopener noreferrer"
         >
           View on Github
         </a>
-        <a 
-          href={"mailto:" + data.email} 
-          className="button-link" 
+        <a
+          href={`mailto:${data.lead.email}?subject=Request to join ${project.name}`}
```
Comment (partiallymisplaced) [inline src/components/ProjectCard.tsx]: 👍

File: src/contexts/AppContext.tsx [hunk 4]
```diff
@@ -20,29 +22,48 @@ export const AppContextProvider = (props) => {
...
+  }
+
+  const fetchUserData = async (token) => {
+    const result = await fetch('/user/', token);
     setUser(result.data.user);
+  };
+
+  const fetchProjectData = async (token) => {
+    const result = await fetch('/all/', token);
+    console.log("xx fetchProjectData", result);
```
Comment (partiallymisplaced) [inline src/contexts/AppContext.tsx]: 🗑

File: src/contexts/AppContext.tsx [hunk 5]
```diff
@@ -20,29 +22,48 @@ export const AppContextProvider = (props) => {
...
     setProjects(result.data.projects);
+    console.log("xx after set projects", projects);
```
Comment (partiallymisplaced) [inline src/contexts/AppContext.tsx]: 🗑

File: src/pages/Projects.tsx [hunk 6]
```diff
@@ -4,6 +4,7 @@ import ProjectCard from "../components/ProjectCard";
 import { AppContext } from "../contexts/AppContext";
 import { Project } from "../types";
 import "../style/pages/Projects.css";
+import { throwStatement } from "@babel/types";
```
Comment (partiallymisplaced) [inline src/pages/Projects.tsx]: Is that used?

File: src/pages/Token.tsx [hunk 7]
```diff
@@ -1,20 +1,25 @@
...
-export function Token(props) {
-  const [cookies, setCookie] = useCookies(['auth_token']);
-  setCookie('auth_token', props.match.params.token, {
-    maxAge: 60 * 60 * 24 * 30, // 1 month
-    path: '/'
-  });
-  var auth_token = cookies.auth_token;
+export function Token(props: TokenProps) {
 
+  console.log("VVVVV cookies", Cookies.get());
```
Comment (partiallymisplaced) [inline src/pages/Token.tsx]: 🗑

Comment (partiallymisplaced) [reply src/pages/Token.tsx]: Also line 12, 15, 21

File: src/components/ProjectCard.tsx (hunk 2 above)
Comment (nurmerey) [reply src/components/ProjectCard.tsx]: It will open in a new tab

File: src/pages/Projects.tsx (hunk 6 above)
Comment (nurmerey) [reply src/pages/Projects.tsx]: Removed

File: src/components/ProjectCard.tsx (hunk 2 above)
Comment (partiallymisplaced) [reply src/components/ProjectCard.tsx]: 👍

File: src/pages/Projects.tsx (hunk 6 above)
Comment (partiallymisplaced) [reply src/pages/Projects.tsx]: 👍

**Pull request merged**